### DISCUSSION TITLE
How to create function which can access and use as type.

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3191] [Q&A]
I want to expose function from element so that user can use that call. How I can set type for that?

**Comment by sorvell** [comment]
Are you trying to expose a method or a method that takes a function as an argument? You'll need to explain more what you're trying to do and wha the problem is.

In either case, consulting the TypeScript documentation is likely the best resource: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#functions

**Reply by jitendraP-ashutec** [reply]
@sorvell  I want to expose a method so that user can call that function.

Like ` <my-element  id="myElement"></my-element>`

and now I want to call function of my-elelement
Like: `myElement.myElementEvent(param1,param2)`


So I want to support type for this function

**Comment by dbwodlf3** [comment]
You can just call the method.
If you want to guide type hinting, then you need to use typescript or jsdoc.

Example)
```html
<my-element id="MyElement"></my-element>
```

```ts
const my_element = document.getElementById("MyElement");

my_element.callMyMethod();
```

**Reply by jitendraP-ashutec** [reply]
Thanks for answer @dbwodlf3 , I can able to access as you suggested, but my question is how can I define as type `calMyMethod()` so that its suggest in typescript
